packages/entities/entities-shared/src/composables/useTruncationDetector.ts

```diff
@@ -0,0 +1,33 @@
+import { ref, watch, computed } from 'vue'
+import type { Ref } from 'vue'
+
+/**
+ * Returns a computed boolean indicating whether the textContent is truncated.
+ *
+ * @param textContent {Ref<HTMLElement>} ref to the element to detect truncation on.
+ *
+ * @returns {isTruncated: Ref<boolean>} ref to a boolean indicating whether the textContent is truncated.
+ */
+export default function useTruncationDetector(textContent: Ref<HTMLElement>) {
+  const textOffsetWidth = ref(0)
+  const textScrollWidth = ref(0)
+  const truncationCalculated = ref(false)
+
+  const detectTruncation = () => {
+    if (textContent.value && !truncationCalculated.value) {
+      textOffsetWidth.value = (textContent.value as HTMLElement).offsetWidth
+      textScrollWidth.value = (textContent.value as HTMLElement).scrollWidth
+      truncationCalculated.value = true
+    }
+  }
+
+  watch(textContent, detectTruncation)
```

Comment:
Since you're using a watcher here, it would be better to utilize the data initialized internally:
```suggestion
  const detectTruncation = () => {
    if (textContent.value && !truncationCalculated.value) {
      textOffsetWidth.value = (textContent.value as HTMLElement).offsetWidth
      textScrollWidth.value = (textContent.value as HTMLElement).scrollWidth
      truncationCalculated.value = true
    }
  }

  watch(textContent, (content: HTMLElement) => {
    if (content && !truncationCalculated.value) {
      textOffsetWidth.value = content.offsetWidth
      textScrollWidth.value = content.scrollWidth
      truncationCalculated.value = true
    }
  })
```